tools/block-generator/generator/generate.go

```diff
@@ -369,90 +299,59 @@ func (g *generator) WriteBlock(output io.Writer, round uint64) error {
 		return nil
 	}
 	// round == nextRound case
+
 	err := g.startRound()
 	if err != nil {
 		return err
 	}
-	numTxnForBlock := g.txnForRound(g.round)
+	if g.round == 0 {
+		fmt.Printf("initializing generator with starting txnCounter from the genesis block: %d\n", g.txnCounter)
```

Comment:
since this is in WriteBlock(),
```suggestion
		fmt.Printf("starting txnCounter: %d\n", g.txnCounter)
```